service.go

```diff
@@ -35,51 +60,113 @@ type Config struct {
+	// (Optional). Defaults to a no op logger.
+	Logger log.Logger
+
+	// MetricsProvider for instrumenting this package.
+	// Gets passed to Argus config before initializing the client.
+	// (Optional). Defaults to a no op provider.
+	MetricsProvider provider.Provider
 }
 
 type service struct {
-	argus   *chrysom.Client
-	loggers *loggerGroup
-	config  Config
+	argus  chrysom.PushReader
+	logger log.Logger
+	config Config
 }
 
 func (s *service) Add(owner string, w Webhook) error {
 	item, err := webhookToItem(w)
 	if err != nil {
-		return err
+		return fmt.Errorf(errFmt, errFailedWebhookConversion, err)
 	}
 	result, err := s.argus.PushItem(item.ID, s.config.Bucket, owner, item)
 	if err != nil {
-		return err
+		return fmt.Errorf(errFmt, errFailedWebhookPush, err)
 	}
 
 	if result == chrysom.CreatedPushResult || result == chrysom.UpdatedPushResult {
 		return nil
 	}
-	return errors.New("operation to add webhook to db failed")
+	return fmt.Errorf("%w: %s", errNonSuccessPushResult, result)
 }
 
+// AllWebhooks returns the set of all webhooks associated with the given owner.
+// Note: While webhooks stored through Argus have this item to owner relationship
+// information, those stored through SNS do not have that piece of information.
+// This opens the possibility of not capturing those items that were just migrated
+// from SNS to Argus. For this reason, when migration configuration is provided,
+// AllWebhooks provides a set with data from both the migrated list of webhooks
+// from SNS and those already on Argus.
+// Webhooks are returned sorted by the config.url field. If there is need, we
+// could make this configurable or user-provided in future releases.
 func (s *service) AllWebhooks(owner string) ([]Webhook, error) {
-	s.loggers.Debug.Log("msg", "AllWebhooks called", "owner", owner)
+	webhookSet := make(map[string]Webhook)
+	if s.config.Migration != nil {
+		err := s.captureMigratedWebhooks(webhookSet)
+		if err != nil {
+			return nil, fmt.Errorf(errFmt, errFailedMigratedWebhooksFetch, err)
+		}
+	}
+
 	items, err := s.argus.GetItems(s.config.Bucket, owner)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf(errFmt, errFailedWebhooksFetch, err)
 	}
-	webhooks := []Webhook{}
+
 	for _, item := range items {
 		webhook, err := itemToWebhook(item)
 		if err != nil {
-			continue
+			return nil, fmt.Errorf(errFmt, errFailedItemConversion, err)
 		}
+		webhookSet[item.ID] = webhook
+	}
+	webhookList := toSlice(webhookSet)
+	sort.Slice(webhookList, func(i, j int) bool {
+		return webhookList[i].Config.URL < webhookList[j].Config.URL
+	})
+
+	return webhookList, nil
+}
+
+func (s *service) captureMigratedWebhooks(webhookSet map[string]Webhook) error {
+	items, err := s.argus.GetItems(s.config.Migration.Bucket, s.config.Migration.Owner)
+	if err != nil {
+		return err
+	}
+
+	for _, item := range items {
+		webhook, err := itemToWebhook(item)
+		if err != nil {
+			return err
+		}
+		webhookSet[item.ID] = webhook
```

Comment:
What happens when webhookSet is nil? Why not just return a map? It looks like we're passing an empty one anyways.

Reply:
As we're removing all the migration related code, this will no longer be relevant.